AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine.swift

```diff
@@ -131,15 +153,36 @@ class RemoteSyncEngine: RemoteSyncEngineBehavior {
     private func pauseMutations() {
         log.debug(#function)
         outgoingMutationQueue.pauseSyncingToCloud()
+        remoteSyncTopicPublisher.send(.mutationsPaused)
     }
 
     private func setUpCloudSubscriptions(api: APICategoryGraphQLBehavior,
                                          storageAdapter: StorageEngineAdapter) {
         log.debug(#function)
         let syncableModelTypes = ModelRegistry.models.filter { $0.schema.isSyncable }
-        reconciliationQueue = AWSIncomingEventReconciliationQueue(modelTypes: syncableModelTypes,
-                                                                  api: api,
-                                                                  storageAdapter: storageAdapter)
+        reconciliationQueue = reconciliationQueueFactory(syncableModelTypes, api, storageAdapter)
+        reconciliationQueueCancellable = reconciliationQueue?.publisher.sink(receiveCompletion: { completionMode in
+            if case .failure(let error) = completionMode {
+                self.remoteSyncTopicPublisher.send(completion: .failure(error))
+            }
+            if case .finished = completionMode {
+                self.log.error("reconciliationQueue was finished.  This is unexpected")
```

Comment:
Can we add some data/info into the error log that can help in debugging further?

Reply:
The only reason we would hit this case would be if the reconciliationQueue sends a .finished message to the subject. E.g. `modelReconciliationQueueSubject.send(completion: .finished)` -- so really this case is here for completeness for Combine -- we aren't really expecting to hit this code.  I am also hesitant to move these strings to a constant file unless we are using this message in more than 2 places.  In this case, it is a very specific case -- that is- the reconciliationQueue has finished, which is being (only) being managed by the RemoteSyncEngine.  That being said, relooking at the message, i think it makes sense to update message to be clearer for developers.  Thanks for the input.